fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckTelemetryService.kt

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.middleware
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import mozilla.components.browser.state.selector.selectedTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.support.base.log.logger.Logger
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+/**
+ * Service that handles telemetry events for review checker.
+ */
+interface ReviewQualityCheckTelemetryService {
+
+    /**
+     * Sends a click attribution event for a given product aid.
+     */
+    suspend fun recordRecommendedProductClick(productAid: String): Unit?
+
+    /**
+     * Sends an impression attribution event for a given product aid.
+     */
+    suspend fun recordRecommendedProductImpression(productAid: String): Unit?
+}
+
+/**
+ * Service that handles the network requests for the review quality check feature.
+ *
+ * @param browserStore Reference to the application's [BrowserStore] to access state.
+ */
+class DefaultReviewQualityCheckTelemetryService(
+    private val browserStore: BrowserStore,
+) : ReviewQualityCheckTelemetryService {
+
+    private val logger = Logger("ReviewQualityCheckTelemetryService")
+
+    override suspend fun recordRecommendedProductClick(productAid: String) =
+        withContext(Dispatchers.Main) {
```

Comment:
Should we inject `Dispatchers.Main` before use?

Reply:
Would usually make sense but this is quite a straightforward event call that's not been tested independently since it's simply forwarding the call, so injecting here doesn't make much sense IMO. It's a good idea in general!